File: app/src/main/java/org/stepic/droid/core/ILessonSessionManager.java
package org.stepic.droid.core;

import org.jetbrains.annotations.Nullable;
import org.stepic.droid.model.Attempt;
import org.stepic.droid.model.Submission;

public interface ILessonSessionManager {
    @Nullable
    Submission restoreSubmissionForStep(long stepId);

    @Nullable
    Attempt restoreAttemptForStep(long stepId);

    void saveSession(long stepId, @Nullable Attempt attempt, @Nullable Submission submission);
}

File: app/src/main/java/org/stepic/droid/core/ILessonSessionManager.kt
package org.stepic.droid.core

import org.stepic.droid.model.Attempt
import org.stepic.droid.model.Submission

interface ILessonSessionManager {
    fun restoreSubmissionForStep(stepId: Long): Submission?

    fun restoreAttemptForStep(stepId: Long): Attempt?

    fun saveSession(stepId: Long, attempt: Attempt?, submission: Submission?)

    fun reset()
}
